lib/Bio/EnsEMBL/Versioning/Pipeline/CheckCoordinateOverlap.pm

```diff
@@ -0,0 +1,125 @@
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2017] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+=pod
+
+
+=head1 CONTACT
+
+  Please email comments or questions to the public Ensembl
+  developers list at <dev@ensembl.org>.
+
+  Questions may also be sent to the Ensembl help desk at
+  <helpdesk@ensembl.org>.
+
+=head1 NAME
+
+Bio::EnsEMBL::Versioning::Pipeline::CheckCoordinateOverlap
+
+=head1 DESCRIPTION
+
+Checks for transcripts that overlap with ucsc transcripts based on coordinate position
+
+=cut
+
+package Bio::EnsEMBL::Versioning::Pipeline::CheckCoordinateOverlap;
+
+use strict;
+use warnings;
+
+use parent qw/Bio::EnsEMBL::Versioning::Pipeline::Base/;
+
+use Bio::EnsEMBL::Utils::Exception qw/throw/;
+use Bio::EnsEMBL::Versioning::Broker;
+use Bio::EnsEMBL::Mongoose::Persistence::QueryParameters;
+use Bio::EnsEMBL::Mongoose::IndexReader;
+use Bio::EnsEMBL::Mongoose::Serializer::RDF;
+use Bio::EnsEMBL::Mongoose::Serializer::RDFCoordinateOverlap;
+use Bio::EnsEMBL::Mongoose::Taxonomizer;
+use Bio::EnsEMBL::Versioning::CoordinateMapper;
+use Bio::EnsEMBL::ApiVersion qw/software_version/;
+use File::Path qw/make_path/;
+use File::Spec;
+use IO::File;
+use Bio::EnsEMBL::Utils::IO qw/slurp_to_array/;
+
+
+sub fetch_input {
+  my $self = shift;
+  my $eg = $self->param('eg');
+  my $release;
+  if ($eg) {
+     $release = $self->param('eg_version');
+  } else {
+    $release = software_version;
+  }
+  my $species = $self->param('species');
+  
+  # Derive location of coordinate_overlap
+  my $indexer = Bio::EnsEMBL::Mongoose::IndexReader->new(species => $species);
+
+  $self->param('indexer',$indexer);
+  
+  # Define where the RDF output will go and create the folder
+  my $output_path = File::Spec->join( $self->param('base_path'), "xref_rdf_dumps", $self->param('run_id'), $species, 'coordinate_overlap');
+  if (!-d $output_path) {
+    make_path $output_path or die "Failed to create path: $output_path";
+  }
+  $self->param('output_path',$output_path);
+}
+
+sub run {
+  my ($self) = @_;
+
+  my $run_id = $self->param('run_id'); # Comes from hive param stack
+  my $core_dba = $self->get_DBAdaptor('core');
+  
+  my $indexer = $self->param('indexer');
+  my $output_path = $self->param('output_path');
+  my $species= $self->param('species');
+  
+  my $broker = Bio::EnsEMBL::Versioning::Broker->new;
+  my @sources = qw/refseq ucsc/;
```

Comment:
It may prove useful to refactor this into an overlap-source-factory so that critical pipeline decisions are not tucked in amongst the execution